File: src/main/java/com/wikia/tibia/repositories/QuestRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.QuestGateway;
import com.wikia.tibia.objects.Quest;

public class QuestRepository extends WikiObjectRepository {

    public QuestRepository() {
        super(Quest.class, new QuestGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/QuestRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Quest

class QuestRepository : WikiObjectRepository<Quest>(
  wikiObjectClass = Quest::class.java,
  wikiObjectGateway = WikiObjectGateway(Contract.QUESTS)
)
